### PR TITLE
feat: enforcing labels on KafkaConnection

### PR DESCRIPTION
**Pull request by secondsun**

### CODE REVIEW

**Comment by secondsun** [comment]
See https://github.com/redhat-developer/app-services-operator/issues/244

File: source/rhoas/src/test/java/com/openshift/cloud/test/KafkaConnectionControllerTest.java [hunk 1]
```diff
@@ -55,6 +55,8 @@ public void setSecret() {
   public void testKafkaConnectionRequest() {
     var kafkaConnectionRequest = new KafkaConnectionBuilder()
         .withMetadata(new ObjectMetaBuilder().withGeneration(10l).withNamespace("test")
+            .withLabels(Map.of("app.kubernetes.io/component", "external-service",
```
Comment (wtrocki) [inline source/rhoas/src/test/java/com/openshift/cloud/test/KafkaConnectionControllerTest.java:58]: Use constants here

File: source/rhoas/src/main/java/com/openshift/cloud/controllers/AbstractCloudServicesController.java [hunk 2]
```diff
@@ -52,13 +56,37 @@
...
+  private boolean requiresLabelUpdate(T resource) {
+    var updateRequired = false;
+    if (resource instanceof KafkaConnection) {
+      var labels = resource.getMetadata().getLabels();
+
+      if (labels == null) {
+        resource.getMetadata().setLabels(labels = new HashMap());
+      }
+
+      updateRequired = !(labels.containsKey("app.kubernetes.io/component")
```
Comment (wtrocki) [inline source/rhoas/src/main/java/com/openshift/cloud/controllers/AbstractCloudServicesController.java:78]: managed-by can already exist as it was documented before and used in UI. 
So my take is that we should check for `app.kubernetes.io/component` and force two flags for it.

**Review by wtrocki — COMMENTED** [review]
Minor fixes are required, but otherwise, I do see resources updating

**Review by wtrocki — APPROVED** [review]

**Review by wtrocki — APPROVED** [review]

**Pull request merged**